### PR TITLE
Mason 3

### PR DESCRIPTION
**Pull request by mamc3334**
First go around at aos size rescaling. Had to change process in main, imageaos, and aossize. Had so much fun figuring out vectors and static casting.

### CODE REVIEW

File: common/progargs.cpp [hunk 1]
```diff
@@ -45,8 +45,6 @@ void prog_resize(const int argc, char *argv[])
         cerr << "Invalid resize height: " << argv[5] << "\n";
         exit(-1);
     }
-    util_set_new_w(atoi(argv[4]));
```
Comment (mamc3334) [inline common/progargs.cpp:48]: don't need - pass by value

File: common/utility.cpp [hunk 2]
```diff
@@ -4,70 +4,23 @@
...
-                "Operation:   " << "info\n" <<
-                    "Magic Word:  " << metadata.magic_word <<
-                        "Image Size:   " << metadata.width << "x" << metadata.height << "\n" <<
-                            "Max Level:   " << metadata.intensity << "\n";
-
+    string magic_word;
+    int width, height, intensity;
+    imageFile >> magic_word >> width >> height >> intensity;
+    Image_Attributes output = {magic_word, width, height, intensity};
+    validate_metadata(output);
```
Comment (mamc3334) [inline common/utility.cpp:20]: call validate_metadata each time metadata is retrieved

File: imgaos/imageaos.cpp [hunk 3]
```diff
@@ -6,79 +8,34 @@ using namespace std;
 #include "imageaos.hpp"
 #include <iostream>
 
-void aos_maxlevel(int newIntensity)
+void aos_resize(int width, int height)
 {
-    ifstream inputImageFile(getInFile());
```
Comment (mamc3334) [inline imgaos/imageaos.cpp:11]: max level moved to common.

File: imgaos/aossize.cpp
Comment (fbuggy) [inline imgaos/aossize.cpp:1]: This looks fun

File: common/intensityscaling.cpp [hunk 4]
```diff
@@ -7,21 +8,22 @@ using namespace std;
...
     }
-    ofstream outputImageFile(outputFilename);
+    ofstream outputImageFile(getOutFile());
     if(!outputImageFile.is_open()) {
         cerr << "Failed to open output file\n";
         exit(-1);
     }
     string magic_word;
     int width, height, intensity;
     inputImageFile >> magic_word >> width >> height >> intensity;
```
Comment (fbuggy) [inline common/intensityscaling.cpp:24]: Why not use get_image_metadata here? Would it not be the same speed + cleaner?

**Review by fbuggy — CHANGES_REQUESTED** [review]

File: common/intensityscaling.cpp (hunk 4 above)
Comment (mamc3334) [reply common/intensityscaling.cpp:24]: I agree. @100496948 can you make these changes and make a pull request

**Review by 100496948 — APPROVED** [review]

File: common/intensityscaling.cpp (hunk 4 above)
Comment (100496948) [reply common/intensityscaling.cpp:24]: Yeah, I'll work on it. I'm not sure if i'll be able to do it today

**Review by fbuggy — APPROVED** [review]

**Pull request merged**